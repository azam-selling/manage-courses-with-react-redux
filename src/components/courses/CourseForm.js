import React from "react";
import PropTypes from "prop-types";
import TextInput from "../common/TextInput";
import SelectInput from "../common/SelectInput";

const CourseForm = (props) => (
  <form onSubmit={props.onSave}>
    <h2>{props.course.id ? "Edit" : "Add"} course</h2>
    {props.errors && props.errors.savingError && (
      <div className="alert alert-danger">{props.errors.savingError}</div>
    )}
    <TextInput
      name="title"
      label="Title"
      value={props.course.title}
      onChange={props.onChange}
      error={props.errors.title}
    />
    <SelectInput
      name="authorId"
      label="Author"
      value={props.course.authorId || ""}
      defaultOption="Select Author"
      options={props.authors.map((author) => {
        return {
          value: author.id,
          text: author.name,
        };
      })}
      onChange={props.onChange}
      error={props.errors.authorId}
    />
    <TextInput
      name="category"
      label="Category"
      value={props.course.category}
      onChange={props.onChange}
      error={props.errors.category}
    />

    <button type="submit" disabled={props.saving} className="btn btn-primary">
      {props.saving ? "Saving..." : "Save"}
    </button>
  </form>
);

CourseForm.propTypes = {
  course: PropTypes.object.isRequired,
  authors: PropTypes.array.isRequired,
  errors: PropTypes.object.isRequired,
  onSave: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  saving: PropTypes.bool.isRequired,
};

export default CourseForm;
