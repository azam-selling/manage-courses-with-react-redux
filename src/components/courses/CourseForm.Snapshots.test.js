import React from "react";
import CourseForm from "./CourseForm";
import renderer from "react-test-renderer";
import { courses, authors } from "../../../tools/mockData";

it("Should check text of save button to be 'Saving...' when saving", () => {
  const tree = renderer.create(
    <CourseForm
      course={courses[0]}
      authors={authors}
      errors={{}}
      onChange={jest.fn()}
      onSave={jest.fn()}
      saving
    />
  );
  expect(tree).toMatchSnapshot();
});

it("Should check text of save button to be 'Save' when not saving", () => {
  const tree = renderer.create(
    <CourseForm
      course={courses[0]}
      authors={authors}
      errors={{}}
      onChange={jest.fn()}
      onSave={jest.fn()}
      saving={false}
    />
  );
  expect(tree).toMatchSnapshot();
});
