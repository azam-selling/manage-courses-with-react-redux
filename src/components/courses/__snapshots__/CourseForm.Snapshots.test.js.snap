// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should check text of save button to be 'Save' when not saving 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <h2>
    Edit
     course
  </h2>
  <div
    className="form-group"
  >
    <label>
      Title
    </label>
    <div>
      <input
        className="form-control mb-3"
        name="title"
        onChange={[MockFunction]}
        type="text"
        value="Securing React Apps with Auth0"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label>
      Author
    </label>
    <div>
      <select
        className="form-control mb-3"
        name="authorId"
        onChange={[MockFunction]}
        value={1}
      >
        <option
          value=""
        >
          Select Author
        </option>
        console.log(options)
        <option
          value={1}
        >
          John Stuart Mill
        </option>
        <option
          value={2}
        >
          Scott Allen
        </option>
        <option
          value={3}
        >
          Dan Wahlin
        </option>
      </select>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label>
      Category
    </label>
    <div>
      <input
        className="form-control mb-3"
        name="category"
        onChange={[MockFunction]}
        type="text"
        value="JavaScript"
      />
    </div>
  </div>
  <button
    className="btn btn-primary"
    disabled={false}
    type="submit"
  >
    Save
  </button>
</form>
`;

exports[`Should check text of save button to be 'Saving...' when saving 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <h2>
    Edit
     course
  </h2>
  <div
    className="form-group"
  >
    <label>
      Title
    </label>
    <div>
      <input
        className="form-control mb-3"
        name="title"
        onChange={[MockFunction]}
        type="text"
        value="Securing React Apps with Auth0"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label>
      Author
    </label>
    <div>
      <select
        className="form-control mb-3"
        name="authorId"
        onChange={[MockFunction]}
        value={1}
      >
        <option
          value=""
        >
          Select Author
        </option>
        console.log(options)
        <option
          value={1}
        >
          John Stuart Mill
        </option>
        <option
          value={2}
        >
          Scott Allen
        </option>
        <option
          value={3}
        >
          Dan Wahlin
        </option>
      </select>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label>
      Category
    </label>
    <div>
      <input
        className="form-control mb-3"
        name="category"
        onChange={[MockFunction]}
        type="text"
        value="JavaScript"
      />
    </div>
  </div>
  <button
    className="btn btn-primary"
    disabled={true}
    type="submit"
  >
    Saving...
  </button>
</form>
`;
